src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -215,32 +282,19 @@ void WatchHierarchy(IVsHierarchy hierarchyToWatch)
                 }
 
                 // We may have multiple projects with the same hierarchy, but we can use __VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext to distinguish
+                mappedHierarchy = hierarchy;
```

Comment:
Is it easier just to do this right away and then let the rest of the function universally use mappedHierarchy? Or should we just pass the hierarchy ref instead of having one in parameter and one out parameter both of which are (hopefully) in sync?

Reply:
so, i dont' understand this code at all.  my primary goal was preserving the semantics of hte prior system (which seems super compplex to me) :)